open-craft/xblock-poll
# Ensure continuous support of Django 3.2 and celery v5.0+

## Description

This PR ensures that `xblock-poll` supports

- Python 2.7 - Django 1.11
- Python 3.6 - Django 2.2
- Python 3.6 - Django 3.2
- Python 3.8 - Django 2.2
- Python 3.8 - Django 3.2

Also

- it replaces the CI/CD pipeline with CircleCI
- makes some small adjustments on the README file for better readability
- introduces `tox` for testing
- adds trove classifiers to `setup.py`
- bumps package version
- adds missing translations that were committed
- adds support for celery v5.0+

## Supporting information

* Related edX [GitHub Issue](https://github.com/edx/upgrades/issues/73)

Celery did remove the `celery.decorators` module, or at least `tasks` is not part of that anymore. Therefore, mimicking the previous behaviour, we have to import the current Celery app and use that task. OpenCraft currently upgrading Celery on the edX platform as well, so we wouldn't need to bump xblock-poll twice in the platform:

* https://github.com/edx/configuration/pull/6586
* https://github.com/edx/edx-platform/pull/29046

### Dependencies

* N/A

## Testing instructions

* Install necessary requirements
* Run `tox` -- get a coffee, it will take a while because of bok-choy
* Go to the test instance and check [this poll](https://apps.pr29046.sandbox.opencraft.hosting/learning/course/course-v1:edX+DemoX+Demo_Course/block-v1:edX+DemoX+Demo_Course+type@sequential+block@a8d8421a98074e8397e41dd5bff651ce/block-v1:edX+DemoX+Demo_Course+type@vertical+block@3328699da07f4f7c9e8413d50e7646fe) and survey to validate it is working even on master
* Export results

## Deadline

ASAP

## Reviewers

@arbrandes
